### PR TITLE
Add support for Cross joins

### PR DESCRIPTION
**Pull request by JamesRTaylor**

### CODE REVIEW

File: isthmus/src/test/java/io/substrait/isthmus/ProtoPlanConverterTest.java [hunk 1]
```diff
@@ -57,6 +63,42 @@ public void filter() throws IOException, SqlParseException {
...
+        assertProtoPlanRoundrip(
+            "select\n"
+                + "  c.c_custKey,\n"
+                + "  o.o_custkey\n"
+                + "from\n"
+                + "  \"customer\" c cross join\n"
+                + "  \"orders\" o",
+            new SqlToSubstrait(
+                new SqlToSubstrait.Options(new SubstraitRelVisitor.Options(KEEP_AS_CROSS_JOIN))));
+    plan1.getRoots().forEach(t -> t.getInput().accept(visitor));
```
Comment (jinfengni) [inline isthmus/src/test/java/io/substrait/isthmus/ProtoPlanConverterTest.java:86]: nice way to verify `CrossRel` is present in the plan.

**Comment by jacques-n** [comment]
Need some conflict resolution here.

**Comment by JamesRTaylor** [comment]
Rebased, @jacques-n .

**Review by jacques-n — APPROVED** [review]
LGTM. Thanks @JamesRTaylor

**Pull request merged**